README.md

```diff
@@ -24,12 +21,9 @@ use aya_log_ebpf::info;
 
 use core::mem;
 use network_types::{
-    l2::ethernet::{EthHdr, ETH_HDR_LEN},
-    l3::{
-        ip::{Ipv4Hdr, IPV4_HDR_LEN},
-        L3Protocol,
-    },
-    l4::{tcp::TcpHdr, udp::UdpHdr, L4Protocol},
+    l2::ethernet::{EthHdr, EthProto, ETH_HDR_LEN},
```

Comment:
personally, at least for this case, I'd prefer to have `ETH_HDR_LEN` exposed so that I could save a line of code with `let offset = mem::size_of::<EthHdr>();`.  

With the pattern above, the current line like:
`let tcphdr: *const TcpHdr = unsafe { ptr_at(&ctx, ETH_HDR_LEN + IPV4_HDR_LEN)? };`
would expand to something like four lines?

Those lines are essentially "boilerplate" [in this case] - personally, I'd prefer to avoid them and have shorter functions.  With the optional and more complex/less "ossified" fields, it could be different - maybe we can show in the docs both patterns?

Reply:
Honestly, I decided to push those constants here, because I was annoyed by having them across multiple projects and having to write those myself every time I'm creating some dummy TC/XDP project for quick testing.

That said, I'm fine with removing those constants from here since we mostly agree on having a separate crate which wraps up the header parsing in general. @dave-tucker you missed that discussion on last meeting, but the plan is that @shaneutt wants to create a crate, based on network-types, which is going to wrap header parsing, so people don't have to do any unsafe operations to parse MAC, IP and ports.